dynamic/discovery/discovery.go

```diff
@@ -125,14 +132,29 @@ func (rm *ResourceMap) refresh() {
 				apiResource.Version = gv.Version
 			}
 			gve.resources[apiResource.Name] = apiResource
-			// Remember how to map back from Kind to resource.
+			// Remember how to map back from Kind/subresource to resource.
 			// This is different from what RESTMapper provides because we already know
 			// the full GroupVersionKind and just need the resource name.
-			// Make sure we don't choose a subresource like "pods/status".
-			if !strings.ContainsRune(apiResource.Name, '/') {
+			// Remember which resources are subresources, and map the kind to the main resource.
+			if strings.ContainsRune(apiResource.Name, '/') {
+				gve.subresources[apiResource.Name] = apiResource
+			} else {
 				gve.kinds[apiResource.Kind] = apiResource
 			}
 		}
+
+		// Group all subresources for a resource.
+		for apiSubresourceName, _ := range gve.subresources {
+			arr := strings.Split(apiSubresourceName, "/")
+			apiResourceName := arr[0]
+			subresourceKey := arr[1]
+			apiResource := gve.resources[apiResourceName]
```

Comment:
Just in case the discovery result is something unexpected, let's check here if `apiResource` is `nil` and skip it if so. We don't want to panic in that case.

Reply:
Good point.
```
if apiResource == nil {
    continue
}
```